Clean/Circuit/Basic.lean

```diff
@@ -613,21 +619,32 @@ attribute [circuit_norm] Vector.map_mk List.map_toArray List.map_cons List.map_n
 attribute [circuit_norm] Vector.append_singleton Vector.mk_append_mk Vector.push_mk
   Array.append_singleton Array.append_empty List.push_toArray
   List.nil_append List.cons_append List.append_toArray
-  Vector.mapRange_zero Vector.mapRange_succ Vector.toArray_push Array.push_toList List.append_assoc
+  Vector.toArray_push Array.push_toList List.append_assoc
   Vector.eq_mk Vector.mk_eq
 
+-- `getElem` lemmas should be tried before expanding Vectors/Lists
+attribute [circuit_norm ↓] Fin.getElem_fin
+  Vector.getElem_map Vector.getElem_mapFinRange Vector.getElem_mapRange Vector.getElem_finRange
+  Vector.getElem_push Vector.getElem_set Vector.getElem_cast
+  Vector.getElem_mk Vector.getElem_toArray
+  List.getElem_cons_zero List.getElem_cons_succ List.getElem_toArray
+
+/-
+lemmas that would expand `Vector.{mapRange, mapFinRange}` are not added to the simp set,
+because they would sometimes be applied too eagerly where using the corresponding `getElem` lemma is much better
+-/
+-- attribute [circuit_norm] Vector.mapRange_zero Vector.mapRange_succ Vector.mapFinRange_succ Vector.mapFinRange_zero
```

Comment:
these lemmas led to some really annoying terms in the keccak round proof
I asked about it on Zulip: [#general > ✔ how to tweak simp set to not unfold the &#96;list&#96; in &#96;list...](https://leanprover.zulipchat.com/#narrow/channel/113488-general/topic/.E2.9C.94.20how.20to.20tweak.20simp.20set.20to.20not.20unfold.20the.20.60list.60.20in.20.60list.2E.2E.2E/with/517759673)

but the conclusion is that this annoying simplifications can't be avoided. I think it's better err on the side of simping too little than too much, so I removed them from the set